components/base/kinematicbase/differentialDrive.go

```diff
@@ -36,44 +37,42 @@ var ErrMovementTimeout = errors.New("movement has timed out")
 func wrapWithDifferentialDriveKinematics(
 	ctx context.Context,
 	b base.Base,
+	logger golog.Logger,
 	localizer motion.Localizer,
 	limits []referenceframe.Limit,
 	maxLinearVelocityMillisPerSec float64,
 	maxAngularVelocityDegsPerSec float64,
 ) (KinematicBase, error) {
+	ddk := &differentialDriveKinematics{
+		Base:                          b,
+		logger:                        logger,
+		localizer:                     localizer,
+		maxLinearVelocityMillisPerSec: maxLinearVelocityMillisPerSec,
+		maxAngularVelocityDegsPerSec:  maxAngularVelocityDegsPerSec,
+	}
+
 	geometries, err := b.Geometries(ctx)
 	if err != nil {
 		return nil, err
 	}
-
-	var geometry spatialmath.Geometry
 	if len(geometries) > 0 {
-		geometry = geometries[0]
-	}
-	model, err := referenceframe.New2DMobileModelFrame(b.Name().ShortName(), limits, geometry)
-	if err != nil {
-		return nil, err
+		ddk.geometry = geometries[0]
 	}
 
-	fs := referenceframe.NewEmptyFrameSystem("")
-	if err := fs.AddFrame(model, fs.World()); err != nil {
+	ddk.model, err = referenceframe.New2DMobileModelFrame(b.Name().ShortName(), limits, ddk.geometry)
+	ddk.fs = referenceframe.NewEmptyFrameSystem("")
+	if err := ddk.fs.AddFrame(ddk.model, ddk.fs.World()); err != nil {
 		return nil, err
 	}
-
-	return &differentialDriveKinematics{
-		Base:                          b,
-		localizer:                     localizer,
-		model:                         model,
-		fs:                            fs,
-		maxLinearVelocityMillisPerSec: maxLinearVelocityMillisPerSec,
-		maxAngularVelocityDegsPerSec:  maxAngularVelocityDegsPerSec,
-	}, nil
+	return ddk, nil
 }
 
 type differentialDriveKinematics struct {
 	base.Base
+	logger                        golog.Logger
 	localizer                     motion.Localizer
-	model                         referenceframe.Model
+	model                         referenceframe.Frame
+	geometry                      spatialmath.Geometry
```

Comment:
Maybe this will become clear as I read the rest of the PR, but doesn't this duplicate the geometry between both this field and the model?

Reply:
no thats a good catch, nice one